test/common/ssl/ssl_socket_test.cc

```diff
@@ -563,6 +564,87 @@ TEST_P(SslSocketTest, NoCert) {
            GetParam());
 }
 
+// Multiple RSA certificates are rejected.
+TEST_P(SslSocketTest, AtMostOneRsaCert) {
+  const std::string client_ctx_yaml = R"EOF(
+    common_tls_context:
+  )EOF";
+
+  const std::string server_ctx_yaml = R"EOF(
+  common_tls_context:
+    tls_certificates:
+    - certificate_chain:
+        filename: "{{ test_rundir }}/test/common/ssl/test_data/selfsigned_cert.pem"
+      private_key:
+        filename: "{{ test_rundir }}/test/common/ssl/test_data/selfsigned_key.pem"
+    - certificate_chain:
+        filename: "{{ test_rundir }}/test/common/ssl/test_data/selfsigned2_cert.pem"
+      private_key:
+        filename: "{{ test_rundir }}/test/common/ssl/test_data/selfsigned_key.pem"
+)EOF";
+
+  EXPECT_THROW_WITH_REGEX(testUtil(client_ctx_yaml, server_ctx_yaml, "", "", "", "", "", "", "",
+                                   "ssl.no_certificate", true, GetParam()),
```

Comment:
The stat and connected status seem bogus here, since this will throw before reaching that stage.

Maybe consider moving those "bad configuration" tests to `test/common/ssl/context_impl_test.cc`, where we usually test them?